packages/lib/services/plugins/PluginService.ts

```diff
@@ -142,17 +142,20 @@ export default class PluginService extends BaseService {
 		this.isSafeMode_ = v;
 	}
 
-	public waitForLoadedPluginsChange() {
-		return new Promise<void>(resolve => {
-			this.pluginsChangeListeners_.push(() => resolve());
-		});
+	public addLoadedPluginsChangeListener(listener: ()=> void) {
+		this.pluginsChangeListeners_.push(listener);
+
+		return {
+			remove: () => {
+				this.pluginsChangeListeners_ = this.pluginsChangeListeners_.filter(l => (l !== listener));
+			},
+		};
 	}
```

Comment:
Previously, `waitForLoadedPluginsChange` (added to support mobile plugins), was used to wait for changes to the list of loaded plugins in a `while` loop. For example,
```ts
while (!event.cancelled) {
   await service.waitForLoadedPluginsChange();
   ... update something ...
}
```
This, however, has the drawback that `waitFor` listeners can't be cancelled when the plugin unloads. It's instead necessary to wait for the listener to be fired, before it can be removed. Additionally, it's necessary to call `waitForLoadedPluginsChange` from async code.

Because `addLoadedPluginsChangeListener` is now used in more places (`usePlugin`), it has been refactored.